packages/kbn-observable/README.md

```diff
@@ -0,0 +1,198 @@
+# `@kbn/observable`
+
+`@kbn/observable` is an observable library based on the [proposed
+`Observable`][proposal] feature and ideas from [RxJS][rxjs].
+
+Why build our own observable library? The main reason is that we don't want to
+tie our plugin apis heavily to a large dependency, but rather expose something
+that's much simpler and that we have control over ourselves.
+
+In addition, `@kbn/observable` implements `System.observable` which enables
+interop between observable libraries, which means plugins can use whatever
+observable library they want if they don't want to rely on `@kbn/observable`.
+
+## Example
+
+```js
+import { $of, map, last } from '@kbn/observable';
+
+const source = $of(1, 2, 3);
+
+source.pipe(map(i => 2017 + i), last()).subscribe(console.log); // logs 2020
+```
+
+## Just getting started with Observables?
+
+If you are just getting started with observables, a great place to start is with
+Andre Staltz' [The introduction to Reactive Programming you've been
+missing][staltz-intro], which is a great introduction to the ideas and concepts.
+
+The ideas in `@kbn/observable` are heavily based on [RxJS][rxjs], so the [RxJS
+docs][rxjs-docs] are also a good source of introduction to observables and how
+they work in this library.
+
+**NOTE**: Do you know about good articles, videos or other resources that does a
+great job at explaining observables? Add them here so it becomes easier for the
+next person to learn about them!
+
+## Factories
+
+Factories take arguments and produce an observable. Different factories are
+useful for different things, and many behave just like the static functions
+attached to the `Rx.Observable` class in RxJS.
+
+The available factories:
+
+* `$bindCallback` - takes a function as a parameter, calls this function, and
+  emits the value returned from it as its single emission
```

Comment:
Do you have a use case in mind for this operator? I can't think of a case where I'd use it instead of something like `Observable.defer()`

Reply:
> takes a function as a parameter, calls this function, and emits the value returned from it as its single emission

That feels inaccurate. Maybe something like "takes a function as a parameter, calls this function passing it a callback and emits the first argument passed in the first call to the callback as its single emission."

Reply:
I removed it and added `$onSubscribe` instead (similar to `defer` in RxJS).